packages/components/radio/src/RadioGroupProvider.tsx

```diff
@@ -0,0 +1,14 @@
+import { ReactNode, useMemo } from 'react'
+
+import { RadioVariantsProps } from './Radio.variants'
+import { RadioGroupContext } from './RadioGroupContext'
+
+export interface RadioGroupProviderProps extends Pick<RadioVariantsProps, 'intent' | 'size'> {
+  children: ReactNode
+}
+
+export const RadioGroupProvider = ({ intent, size, children }: RadioGroupProviderProps) => {
```

Comment:
perhaps a naive question (😅), but is a Provider necessary? In the Radix documentation examples, I don't see one being used

Reply:
I think that it is. See this example:
```jsx
import React from 'react';
import * as RadioGroup from '@radix-ui/react-radio-group';

const RadioGroupDemo = () => (
  <form>
    <RadioGroup.Root
      className="flex flex-col gap-2.5"
      defaultValue="default"
      aria-label="View density"
    >
      <div className="flex items-center">
        <RadioGroup.Item
          className="bg-white w-[25px] h-[25px] rounded-full shadow-[0_2px_10px] shadow-blackA7 hover:bg-violet3 focus:shadow-[0_0_0_2px] focus:shadow-black outline-none cursor-default"
          value="default"
          id="r1"
        >
          <RadioGroup.Indicator className="flex items-center justify-center w-full h-full relative after:content-[''] after:block after:w-[11px] after:h-[11px] after:rounded-[50%] after:bg-violet11" />
        </RadioGroup.Item>
        <label className="text-white text-[15px] leading-none pl-[15px]" htmlFor="r1">
          Default
        </label>
      </div>
      <div className="flex items-center">
        <RadioGroup.Item
          className="bg-white w-[25px] h-[25px] rounded-full shadow-[0_2px_10px] shadow-blackA7 hover:bg-violet3 focus:shadow-[0_0_0_2px] focus:shadow-black outline-none cursor-default"
          value="comfortable"
          id="r2"
        >
          <RadioGroup.Indicator className="flex items-center justify-center w-full h-full relative after:content-[''] after:block after:w-[11px] after:h-[11px] after:rounded-[50%] after:bg-violet11" />
        </RadioGroup.Item>
        <label className="text-white text-[15px] leading-none pl-[15px]" htmlFor="r2">
          Comfortable
        </label>
      </div>
      <div className="flex items-center">
        <RadioGroup.Item
          className="bg-white w-[25px] h-[25px] rounded-full shadow-[0_2px_10px] shadow-blackA7 hover:bg-violet3 focus:shadow-[0_0_0_2px] focus:shadow-black outline-none cursor-default"
          value="compact"
          id="r3"
        >
          <RadioGroup.Indicator className="flex items-center justify-center w-full h-full relative after:content-[''] after:block after:w-[11px] after:h-[11px] after:rounded-[50%] after:bg-violet11" />
        </RadioGroup.Item>
        <label className="text-white text-[15px] leading-none pl-[15px]" htmlFor="r3">
          Compact
        </label>
      </div>
    </RadioGroup.Root>
  </form>
);

export default RadioGroupDemo;
```

You may have elements between the `RadioGroup` and `Radio` and I want `intent` and `size` to be available

Reply:
Ah, I see now. Instead of requiring style-related props to be set for each instance of RadioGroup items, you prefer to configure once at the RadioGroup root level, hence the React Context usage 👍

Reply:
Yes! you have both options so this way is fully flexible